The CI fails when it build Che for the second time

https://ci.centos.org/view/Devtools/job/devtools-build-run-che-build-master/412/

> [ERROR] Failed to execute goal org.apache.maven.plugins:maven-patch-plugin:1.2:apply (patch-sources) on project fabric8-ide-assembly-ide-war: Patch command failed with exit code 1 for src.main.webapp.IDE.jsp.patch. Please see console and debug output for more information. -> [Help 1]

It works if we `mvn clean install`, but the side effect is that the previous build artifat (with dashboard) is deleted and won't be pushed to the docker hub.

One workaround may be to always `mvn clean install` and build/push the docker image of the corresponding artifact right after (and executing `mvn clean install` only when the docker image has been built). But that may be complicated with current build scripts.

--- a/builds/fabric8-che/assembly/assembly-ide-war/pom.xml
+++ b/builds/fabric8-che/assembly/assembly-ide-war/pom.xml
@@ -243,19 +243,17 @@
                             </tasks>
                         </configuration>
                     </execution>
+                </executions>
+            </plugin>
+            <plugin>
+                <artifactId>maven-clean-plugin</artifactId>
+                <executions>
                     <execution>
-                        <id>clean generated sources in assembly-ide-war</id>
+                        <id>auto-clean</id>
                         <phase>initialize</phase>
                         <goals>
-                            <goal>run</goal>
+                            <goal>clean</goal>
                         </goals>
-                        <configuration>
-                            <target>
-                                <!-- To bypass a problem in the build when 
-                                    rebuilding without the clean option. -->
-                                <delete dir="${project.build.directory}/generated-sources" verbose="true" />
-                            </target>
-                        </configuration>
                     </execution>
                 </executions>
             </plugin>

--- a/cico_do_build_che.sh
+++ b/cico_do_build_che.sh
@@ -24,7 +24,7 @@ mvnche() {
 }
 
 mkdir $NPM_CONFIG_PREFIX 2>/dev/null
-mvnche -B $* install
+mvnche -B $* clean install
 if [ $? -ne 0 ]; then
   echo "Error building che/rh-che with dashboard"
   exit 1;
@@ -32,10 +32,9 @@ fi
 
 if [ "$DeveloperBuild" != "true" ]
   then
-    echo "Build without dashboard is currently skipped because the CI fails (c.f. https://github.com/redhat-developer/rh-che/issues/318). And by the way it's not used in OSIO yet."
-    # mvnche -B -DwithoutDashboard $* install
-    # if [ $? -ne 0 ]; then
-    #   echo "Error building che/rh-che without dashboard"
-    #   exit 1;
-    # fi
+    mvnche -B -DwithoutDashboard -pl=:fabric8-ide-assembly-ide-war,:fabric8-ide-assembly-main $* install
+    if [ $? -ne 0 ]; then
+      echo "Error building che/rh-che without dashboard"
+      exit 1;
+    fi
 fi

